src/main/java/org/broadinstitute/hellbender/engine/ProgressMeter.java

```diff
@@ -0,0 +1,200 @@
+package org.broadinstitute.hellbender.engine;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.function.LongSupplier;
+
+/**
+ * A basic progress meter to print out the number of records processed (and other metrics) during a traversal
+ * at a configurable interval.
+ *
+ * Clients set the update interval at construction, which controls how many records we must process
+ * before printing an update. Then call {@link #start} at traversal start, {@link #update(SimpleInterval)}
+ * after processing each record from the primary input, and {@link #stop} at traversal end to print
+ * summary statistics.
+ *
+ * All output is made at INFO level via log4j.
+ */
+public class ProgressMeter {
```

Comment:
make class final or document which methods are to be extended or overriden and in what way. I suggest just making final to make the intentions clear.